app/controllers/api/user/subscription_targets_controller.rb

```diff
@@ -8,61 +8,36 @@ class SubscriptionTargetsController < ApiController
       before_action :authenticate!
       before_action :find_subscription
 
-      def show
+      def index
         return render json: { message: 'No token was provided' }, status: :unauthorized if @subscription.nil?
 
-        render json: serialize(targets_with_commodities)
-      end
-
-      private
-
-      def targets_with_commodities
-        type = filter_params[:active_commodities_type]&.to_sym
-
-        return [@subscription.subscription_targets.map, @subscription.subscription_targets.size] if type.blank?
-
-        TimeMachine.now do
-          service = Api::User::ActiveCommoditiesService.new(@subscription)
-          commodities, total =
-            if service.respond_to?("#{type}_commodities")
-              service.public_send("#{type}_commodities", page: current_page, per_page: per_page)
-            else
-              [[], 0]
-            end
-          targets = apply_commodities_to_subscription_targets(commodities)
-          [targets, total]
+        case @subscription.subscription_type.name
+        when 'my_commodities'
+          filtered_targets = targets_filter_service.new(@subscription,
```

Comment:
I'm not sure the targets_filter_service method is needed as you are already specifying which subscription type this applies to. I think it's OK to hardcode it here.